eclipse/sw360
# fix(REST): Save otherLicenseIds while patching Release - 1735

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

1. Fix the `otherLicenseIds` not being saved while patching release via REST API issue.
2. Updated the REST API documentation accordingly.

Issue: #1735 


### How To Test?
Update the Release via REST API with following scenario:

1. Add valid `otherLicenseIds` (present in sw360 db) field in request body and update the release -> Update should be successful and `otherLicenseIds` should contain all the licenses added in update request.
2. Add empty `otherLicenseIds` field in request body and update the release -> Update should be successful and `otherLicenseIds` field should be empty.
3. Add invalid (not present in sw360 db) `otherLicenseIds` field in request body and update the release -> Update should fail with error: `License with ids [license-id1, license-id2] does not exist in SW360 database.`


### Checklist
Must:

Signed-off-by: akapti <abdul.kapti@siemens-healthineers.com>
